### PR TITLE
Gherkin Scripts for epic ODC-4645

### PR DESCRIPTION
**Pull request by gajanan-more**
Fixes: https://issues.redhat.com/browse/ODC-4845

Problem: Gherkin Scripts for scenarios in [ODC-4645](https://issues.redhat.com/browse/ODC-4645)

Solution: These are the gherkin scripts designed covering all the scenarios related to epic [ODC-4645](https://issues.redhat.com/browse/ODC-4645)

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    Scenario: Event Sources Filter
+        Given user has created knative service "hello-openshift"
+        And user has created ApiServer Source
+        And user has created Ping Source
+        And user has created Sink Binding
+        And user has created Container Source
+        And user is at Event Sources tab
+        When user clicks on Filter dropdown
+        Then user will see digit "1" in front of ApiServer, Ping, Sink, Container Sources checkboxes
+        And user will see only ApiServer source when checked checkbox in fornt of ApiServer Source
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:56]:
```suggestion
        And user will see only ApiServer source when checked the checkbox in front of ApiServer Source
```

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        And user will see only Container source when checked checkbox in fornt of Container Source
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:57]: *front

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        And user will see only Ping source when checked checkbox in fornt of Ping Source
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:58]: *front

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature (hunk 1 above)
Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:56]: These are the actions. Do yu think they are supposed to be in `Then`

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        And user will see only Sink source when checked checkbox in fornt of Sink Source
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:59]:
```suggestion
        And user will see only Sink source when checked the checkbox in front of Sink Source
```

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    @regression
+    Scenario: Event Sources Filter
+        Given user has created knative service "hello-openshift"
+        And user has created Default Memory Channel
+        And user has created In Memory Channel
+        And user is at Channels tab
+        When user clicks on Filter dropdown
+        Then user will see digit "1" in front of Channel and In Memory Channle checkboxes
+        And user will see only Channel when checked checkbox in fornt of Channel
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:70]:
```suggestion
        And user will see only Channel when checked checkbox in front of Channel
```

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        
+
+    @regression
+    Scenario: Event Sources Filter
+        Given user has created knative service "hello-openshift"
+        And user has created Default Memory Channel
+        And user has created In Memory Channel
+        And user is at Channels tab
+        When user clicks on Filter dropdown
+        Then user will see digit "1" in front of Channel and In Memory Channle checkboxes
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:69]:
```suggestion
        Then user will see digit "1" in front of Channel and In Memory Channel checkboxes
```

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        And user will see only Channel when checked checkbox in fornt of Channel
+        And user will see only In Memory Channel when checked checkbox in fornt of In Memory Channel
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:71]:
```suggestion
        And user will see only In Memory Channel when checked the checkbox in front of In Memory Channel
```

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+    @regression, @manual
+    Scenario: Event Sources Filter
+        Given user has created knative service "hello-openshift"
+        And user has created Github source
+        And user has created Ping Binding
+        And user is at Event Sources tab
+        When user clicks on Filter dropdown
+        Then user will see digit "1" in front of Another Type, Ping Sources checkboxes
+        And user will see only Ping source when checked checkbox in fornt of Ping Source
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:82]:
```suggestion
        And user will see only Ping source when checked the checkbox in front of Ping Source
```

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        And user will see only Another Types of event sources when checked checkbox in fornt of Another Type
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:83]:
```suggestion
        And user will see only Another Types of event sources when checked the checkbox in front of Another Type
```

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature (hunk 1 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:56]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature (hunk 2 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:57]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature (hunk 3 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:58]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature (hunk 4 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:59]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature (hunk 1 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:56]: I thought of it. But then I need to create different scenarios for every checkbox present in that filter drop down which is unnecessary. That's why I arranged it in such a way.

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature (hunk 5 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:70]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature (hunk 7 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:71]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature (hunk 9 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:83]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature (hunk 8 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:82]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature (hunk 6 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:69]: Updated

**Comment by invincibleJai** [comment]
Should also add scenario for details page covering new tab should be should be shown in details for showing associated subscriptions to the selected channel like wise for brokers and triggers

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    
+
+    @regression, @manual
+    Scenario: Event Sources Filter
+        Given user has created knative service "hello-openshift"
+        And user has created Github source
+        And user has created Ping Binding
+        And user is at Event Sources tab
+        When user clicks on Filter dropdown
+        Then user will see digit "1" in front of Another Type, Ping Sources checkboxes
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:81]: Verify this behaviour once , as long as we are able to identify the source we can show that in type. Other type was mentioned as they can't show all types in design cc @sahil143

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+        And user will see only ApiServer source when checked checkbox in front of ApiServer Source
+        And user will see only Container source when checked checkbox in front of Container Source
+        And user will see only Ping source when checked checkbox in front of Ping Source
+        And user will see only Sink source when checked checkbox in front of Sink Source
+        
+
+    @regression
+    Scenario: Event Sources Filter
+        Given user has created knative service "hello-openshift"
+        And user has created Default Memory Channel
```
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:65]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+        Then user will see broker at Brokers tab
+        And user will see Broker Details page after clicking on broker "example"
+
+    
+    @regression @manual
+    Scenario: Triggers tab
+        Given user is at eventing page
+        And user has created Broker "example"
+        And user has created knative service "hello-openshift"
+        And user has created trigger
```
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature:52]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+        And user clicks on Create button
+        Then user will be redirected to Project Details page
+        And user will see channel created
+
+    
+    @regression @manual
+    Scenario: Brokers tab
+        Given user is at eventing page
+        And user has created Broker "example"
+        And user has created knative service "hello-openshift"
```
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature:42]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+        And user selects resource "hello-openshift"
+        And user clicks on Create button
+        Then user will be redirected to Project Details page
+        And user will see ping-source created
+
+    
+    @regression
+    Scenario: Create new Channel
+        Given user is at eventing page
+        And user has created knative service "hello-openshift"
```
[the remark this answers is not in the document]
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature:29]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature (hunk 10 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/filters-Serving-Eventing-Admin.feature:81]: Updated

**Comment by gajanan-more** [comment]
> Should also add scenario for details page covering new tab should be should be shown in details for showing associated subscriptions to the selected channel like wise for brokers and triggers

I have added scenarios for respective details page.

File: frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+
+    
+    @regression, @manual
+    Scenario: Triggers tab
+        Given user is at eventing page
+        And user has created Broker "example"
+        And user has created knative service "hello-openshift"
+        And user has created trigger subscribing to knative service created above
+        Then user will see trigger at Triggers tab
+        And user will see Trigger Details page after clicking on broker "example"
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature:52]: last point i din't got , how will user see `rigger Details page after clicking on broker "example"` ?

File: frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+    @regression, @manual
+    Scenario: Broker Details page
+        Given user is at eventing page
+        And user has created Broker "example"
+        And user has created knative service "hello-openshift"
+        And user has created trigger subscribing to knative service created above
+        And user is at Broker Details page
+        Then user will see Details tab with Broker Details
+        And user will see YAML generated for Broker under YAML tab
+        And user will see Broker name and Subscriber name under Triggers tab
```
Comment (invincibleJai) [inline frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature:83]: See if it make sense can do same for other details scenario

```suggestion
        And user click on broker "example" in broker tab 
        Then user will shown with Broker Details page
        Then user will see Details tab with Broker Details
        And user will see YAML generated for Broker under YAML tab
        And user will see Broker name and Subscriber name under Triggers tab
```

File: frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature (hunk 15 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature (hunk 16 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/Knative/eventingPage-Admin.feature]: Updated

**Pull request merged**